src/all/buondua/src/eu/kanade/tachiyomi/extension/all/buondua/BuonDua.kt

```diff
@@ -77,7 +95,9 @@ class BuonDua() : ParsedHttpSource() {
         chapter.setUrlWithoutDomain(element.select(".is-current").first()!!.attr("abs:href"))
         chapter.chapter_number = 0F
         chapter.name = element.select(".article-header").text()
-        chapter.date_upload = SimpleDateFormat("H:m DD-MM-yyyy", Locale.US).parse(element.select(".article-info > small").text())?.time ?: 0L
+        chapter.date_upload =
+            SimpleDateFormat("H:m DD-MM-yyyy", Locale.US).parse(element.select(".article-info > small").text())?.time
+                ?: 0L
```

Comment:
Move the ```SimpleDateFormat``` to the ```companion object``` or class variable: 
```kt
companion object {
     private val dateFormat = SimpleDateFormat("H:m DD-MM-yyyy", Locale.US)
}
// or
private val dateFormat = SimpleDateFormat("H:m DD-MM-yyyy", Locale.US)
``` 

And use the ```keiyoushi.utils.tryParse```  from utils
```suggestion
        chapter.date_upload = dateFormat.tryParse(element.select(".article-info > small").text())
```

You need to import:
```kt 
import keiyoushi.utils.tryParse
```

https://github.com/keiyoushi/extensions-source/blob/228b657e25788f904b29cde7561b69d5a38f800b/lib-multisrc/mangabox/src/eu/kanade/tachiyomi/multisrc/mangabox/MangaBox.kt#L314